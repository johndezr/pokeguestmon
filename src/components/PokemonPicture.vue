<template>
  <img class="pokemon-image-cover" :class="{'apply-filter': isCover}" :src="image" alt="">
</template>

<script setup lang="ts">
  import { defineProps } from 'vue'

  defineProps<{ image: string | undefined, isCover: boolean }>()
</script>

<style scoped>
  .pokemon-image-cover {
    margin: 20px 0;
    user-select: none;
    -moz-user-select: none;
    -ms-user-select: none;
    -webkit-user-drag: none;
    -webkit-user-select: none;
  }
  .apply-filter {
    filter: brightness(0);
  }
</style>
