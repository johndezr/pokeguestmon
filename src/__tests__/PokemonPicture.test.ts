import { describe, it, expect } from 'vitest'

import { mount } from '@vue/test-utils'
import PokemonPicture from '../components/PokemonPicture.vue'

describe('<PokemonPicture />', () => {
  const defaultProps = {
    image:
      'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/1.png',
    isCover: true,
  }
  const wrapper = mount(PokemonPicture, { props: defaultProps })

  it('renders properly', () => {
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.exists()).toBe(true)
  })
  it('show img with correct source and cover it', () => {
    const img = wrapper.find('img')

    expect(img.exists()).toBe(true)
    expect(img.attributes('src')).toBe(defaultProps.image)
    expect(img.classes()).toContain('apply-filter')
  })
})
