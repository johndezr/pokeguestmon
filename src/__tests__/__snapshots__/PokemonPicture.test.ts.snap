// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<PokemonPicture /> > renders properly 1`] = `"<img data-v-c76ad58e="" class="pokemon-image-cover apply-filter" src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/1.png" alt="">"`;

exports[`HelloWorld > renders properly 1`] = `"<img data-v-c76ad58e="" class="pokemon-image-cover" alt="">"`;
